internal/network/machine_network_cidr.go

```diff
@@ -109,25 +112,20 @@ func VerifyVips(hosts []*models.Host, machineNetworkCidr string, apiVip string,
 		err = VerifyVip(hosts, machineNetworkCidr, ingressVip, "ingress-vip", mustExist, log)
 	}
 	if err == nil {
-		err = verifyDifferentVipAddresses(apiVip, ingressVip)
+		err = VerifyDifferentVipAddresses(apiVip, ingressVip)
 	}
 	return err
 }
 
-func VerifyMachineCIDR(machineCidr string, hosts []*models.Host, log logrus.FieldLogger) error {
+func VerifyMachineCIDR(machineCidr string) error {
 	ip, ipNet, err := net.ParseCIDR(machineCidr)
 	if err != nil {
 		return err
 	}
 	if ipNet.IP.To4().String() != ip.To4().String() {
 		return common.NewApiError(http.StatusBadRequest, errors.Errorf("%s is not a valid machine CIDR", machineCidr))
 	}
-	for _, h := range hosts {
-		if belongsToNetwork(log, h, ipNet) {
-			return nil
-		}
-	}
-	return common.NewApiError(http.StatusBadRequest, errors.Errorf("%s does not belong to any of the host networks", machineCidr))
+	return nil
```

Comment:
The test for ip equality is wrong for IPv6 (line 127). Suggest using ip.Equal(other-ip)